gammapy/analysis/config.py

```diff
@@ -0,0 +1,189 @@
+from astropy.coordinates import Angle
+from astropy.time import Time
+from astropy.units import Quantity
+from gammapy.utils.scripts import make_path, read_yaml
+from pydantic import BaseModel, FilePath, validator
+from pydantic.utils import deep_update
+from pathlib import Path
+from typing import List
+from enum import Enum
+import yaml
+
+__all__ = ["Config", "General"]
+
+
+class AngleType(Angle):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return AngleType(v)
```

Comment:
Can you please add some of the basic tests on what types the config attributes are that we used in the prototype, like here?
https://github.com/cdeil/gammapy-config/blob/422398b002545804bfed91d69c95c96b9cca2d15/test_config_pydantic.py#L12
(one for each custom type you define, use an `isinstance` in the test to check what the config attribute is)

I think here we should `return Angle(v)` to get an angle attribute, not the `AngleType` subclass, no?

Looking at https://pydantic-docs.helpmanual.io/usage/types/#custom-data-types and  https://github.com/samuelcolvin/pydantic/blob/master/pydantic/types.py I don't see them doing type casts, only type checks, so I wonder if we're doing it correctly. For sure `Angle(v)` does what we want, see the tests we had.

Reply:
Ok for the tests.

Re: `return Angle(v)`

When setting the values of an angle (i.e. `config.datasets.geom.wcs.binsize="0.1 deg"`), the type set may be `Angle` or `AngleType` depending on what we decide to return in the validation function. 

Since it seems that type casting does not work on assignment (i.e. default values declared), and it only works on setting, we need to do force casting on the definition of the model. If we do force casting to `AngleType` I would also leave `AngleType` for settings.